blocks/carousel/carousel.js

```diff
@@ -0,0 +1,413 @@
+import { readBlockConfig } from '../../scripts/aem.js';
+import { fetchRecommendations, getMarginActionUrl, mockPredicationConstant } from '../../scripts/mockapi.js';
+
+function updateCarouselView(activeDot) {
+  const dotIndex = parseInt(activeDot.dataset.index, 10);
+  const researchSlider = activeDot.closest('.research-slider');
+  const dots = researchSlider.querySelectorAll('.dot');
+  const currentActiveDot = researchSlider.querySelector('.dot.active');
+  if (currentActiveDot && currentActiveDot.dataset.index === activeDot.dataset.index) {
+    return;
+  }
+  const carouselTrack = researchSlider.querySelector('.carousel-track');
+  const cards = Array.from(carouselTrack.children);
+  const visibleCardsCount = cards.filter((card) => window.getComputedStyle(card).opacity === '1').length;
+  const cardWidth = cards[0].offsetWidth;
+  const moveDistance = dotIndex * cardWidth;
+  cards.forEach((card, index) => {
+    card.style.opacity = (index >= dotIndex && index < dotIndex + visibleCardsCount) ? '1' : '0';
+  });
+
+  carouselTrack.style.transform = `translateX(-${moveDistance}px)`;
+  dots.forEach((dot) => dot.classList.remove('active'));
+  dots[dotIndex].classList.add('active');
+}
+
+function startUpdateCarousel(researchSlider) {
+  const dotsContainer = researchSlider.querySelector('.dots-container');
+  if (!dotsContainer) return; // Exit if dotsContainer doesn't exist
+
+  const dots = dotsContainer.querySelectorAll('.dot');
+  let activeDotIndex = Array.from(dots).findIndex((dot) => dot.classList.contains('active'));
+
+  if (activeDotIndex === -1 || activeDotIndex === dots.length - 1) {
+    return;
+  }
+
+  const intervalId = setInterval(() => {
+    activeDotIndex = (activeDotIndex + 1) % dots.length; // Move to the next dot
```

Comment:
is it really being used? I can see carousel not progressing automatically at https://carousel--icicidirect--aemsites.hlx.live/draft/anagarwa/

Reply:
@shrotia  it works after first load, it progresses to the last slide and then stops. This is what I observed in original site as well .